lib/Internal/functions.php

```diff
@@ -140,6 +144,30 @@ function onCryptoWatchReadability($watcherId, $socket, Deferred $deferred) {
     // If $result is 0, just wait for the next invocation
     if ($result === true) {
         Loop::cancel($watcherId);
+
+        $options = \stream_context_get_options($socket);
+
+        if ($options["ssl"]["verify_peer"] ?? true) {
+            $certs = array_merge([$options["ssl"]["peer_certificate"]], $options["ssl"]["peer_certificate_chain"]);
+
+            foreach ($certs as $cert) {
+                $cert = new Certificate($cert);
+
+                if ($cert->isSelfSigned()) {
+                    // Ignore self-signed certificates, because they don't trust the signature
+                    continue;
+                }
+
+                if (in_array($cert->getSignatureType(), ["md5WithRSAEncryption", "sha1WithRSAEncryption"], true)) {
+                    @\fclose($socket);
+
+                    $deferred->fail(new CryptoException(
+                        "Peer provided a certificate using a MD5 / SHA1 signature scheme"
```

Comment:
`"... which is not considered trustworthy"`?

the message as is implies that the user of the lib knows that MD5/SHA1 certs are not secure.